File: Stage15/C/Problem17103.c
#include <stdio.h>
#include <stdbool.h>
#define MAX 1000000

int main(void) {
    bool isPrime[MAX + 1];
    memset(isPrime, true, sizeof(isPrime));
    isPrime[0] = isPrime[1] = false;

    for (int i = 2; i * i <= MAX; i++) {
        if (isPrime[i]) {
            for (int j = i * i; j <= MAX; j += i) {
                isPrime[j] = false;
            }
        }
    }

    int T;
    scanf("%d", &T);

    while (T--) {
        int N;
        scanf("%d", &N);
        int count = 0;

        if (isPrime[2] && isPrime[N - 2]) count++;

        for (int i = 3; i <= N / 2; i += 2) {
            if (isPrime[i] && isPrime[N - i]) count++;
        }

        printf("%d\n", count);
    }

    return 0;
}

File: Stage15/Python/Problem17103.py
import sys
input = sys.stdin.readline
MAX = 1000000

def main():
    isPrime = [True] * (MAX + 1)
    isPrime[0] = isPrime[1] = False
    
    for i in range(2, int(MAX ** 0.5) + 1):
        if isPrime[i]:
            for j in range(i * i, MAX + 1, i):
                isPrime[j] = False
    
    T = int(input())
    
    for _ in range(T):
        N = int(input())
        count = 0
        
        if isPrime[2] and isPrime[N - 2]:
            count += 1
        
        for i in range(3, N // 2 + 1, 2):
            if isPrime[i] and isPrime[N - i]:
                count += 1
    
        print(count)            
    
if __name__ == "__main__":
    main()